packaging/build/Microsoft.NET.Build.Containers.targets

```diff
@@ -94,6 +89,46 @@
+            <ContainerGenerateLabelsImageSource Condition="'$(ContainerGenerateLabelsImageSource)' == ''">true</ContainerGenerateLabelsImageSource>
+            <ContainerGenerateLabelsImageVersion Condition="'$(ContainerGenerateLabelsImageVersion)' == ''">true</ContainerGenerateLabelsImageVersion>
+            <ContainerGenerateLabelsImageRevision Condition="'$(ContainerGenerateLabelsImageRevision)' == ''">true</ContainerGenerateLabelsImageRevision>
+            <ContainerGenerateLabelsImageVendor Condition="'$(ContainerGenerateLabelsImageVendor)' == ''">true</ContainerGenerateLabelsImageVendor>
+            <ContainerGenerateLabelsImageLicenses Condition="'$(ContainerGenerateLabelsImageLicenses)' == ''">true</ContainerGenerateLabelsImageLicenses>
+            <ContainerGenerateLabelsImageTitle Condition="'$(ContainerGenerateLabelsImageTitle)' == ''">true</ContainerGenerateLabelsImageTitle>
+            <ContainerGenerateLabelsImageBaseDigest Condition="'$(ContainerGenerateLabelsImageBaseDigest)' == ''">true</ContainerGenerateLabelsImageBaseDigest>
+            <ContainerGenerateLabelsImageBaseName Condition="'$(ContainerGenerateLabelsImageBaseName)' == ''">true</ContainerGenerateLabelsImageBaseName>
+        </PropertyGroup>
+
+        <!-- Labels generated from descriptions from the spec at https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys -->
+        <ItemGroup Label="Conventional Label assignment" Condition="'$(ContainerGenerateLabels)' == 'true'">
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageCreated)' == 'true'" Include="org.opencontainers.image.created;org.opencontainers.artifact.created" Value="$([System.DateTime]::UtcNow.ToString('o'))" />
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageDescription)' == 'true' and '$(Description)' != ''" Include="org.opencontainers.artifact.description;org.opencontainers.image.description" Value="$(Description)" />
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageAuthors)' == 'true' and '$(Authors)' != ''" Include="org.opencontainers.image.authors" Value="$(Authors)" />
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageUrl)' == 'true' and '$(PackageProjectUrl)' != ''" Include="org.opencontainers.image.url" Value="$(PackageProjectUrl)" />
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageDocumentation)' == 'true' and '$(PackageDocumentationUrl)' != ''" Include="org.opencontainers.image.documentation" Value="$(PackageDocumentationUrl)" />
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageVersion)' == 'true' and '$(PackageVersion)' != ''" Include="org.opencontainers.image.version" Value="$(PackageVersion)" />
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageVendor)' == 'true' and '$(PackageOwners)' != ''" Include="org.opencontainers.image.vendor" Value="$(PackageOwners)" />
```

Comment:
does `PackageOwners` exist? not mention here https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#pack-target

Reply:
`PackageOwners` similarly doesn't exist - I also would like to run this by @JonDouglas and see if it's just a gap.

Reply:
NuGet's concept of owner is a central registry concept. It is not something MSBuild really should care nor know about. It is assigned at registry time.

https://learn.microsoft.com/en-us/nuget/create-packages/package-authoring-best-practices#package-metadata
https://learn.microsoft.com/en-us/nuget/nuget-org/publish-a-package#manage-package-owners-on-nugetorg

`org.opencontainers.image.vendor` - Name of the distributing entity, organization or individual.

Not sure if this is a static or dynamic field, but if static it would be closest to what `authors` is today as anyone can list in a nuspec as the entity who "authored" the package at that snapshot of time. It does not however represent current ownership.

Reply:
Ok, then we will change our name to something more distinct to containers, since the concepts don't align nicely.